backend-demo/src/screenshot.rs

```diff
@@ -28,9 +28,42 @@ impl ScreenshotImpl for Screenshot {
         _window_identifier: Option<WindowIdentifierType>,
         _options: ScreenshotOptions,
     ) -> Result<ScreenshotResponse> {
-        Ok(ScreenshotResponse::new(
-            url::Url::parse("file:///some/sreenshot").unwrap(),
-        ))
+        let path = xdg_user::pictures().unwrap_or(None);
+        let mut path = match path {
+            Some(p) => std::path::PathBuf::from(p),
+            None => {
+                return Err(PortalError::Failed(format!(
+                    "No XDG pictures directory to save screenshot to"
+                )))
+            }
+        };
+        path.push(format!(
+            "Screenshot_{}.png",
+            chrono::Local::now().format("%Y%m%d_%H%M%S")
+        ));
+        let url = match url::Url::from_file_path(path.as_path()) {
+            Ok(url) => url,
+            _ => {
+                return Err(PortalError::Failed(format!(
+                    "Invalid file path: {}",
+                    path.display()
+                )))
+            }
+        };
+        // The following block is commented out because under Wayland xcap
+        // calls into the org.freedesktop.portal.Screenshot D-Bus API. In a
+        // scenario where this backend demo portal is installed and handles
+        // screenshot requests, it would end up invoking itself recursively.
+        // let monitors = xcap::Monitor::all().unwrap_or(vec![]);
+        // if monitors.is_empty() {
+        // return Err(PortalError::Failed(format!("No monitors found")));
+        // }
+        // let capture = monitors.first().unwrap().capture_image();
+        // let _ = match capture {
+        // Ok(image) => image.save_with_format(path, image::ImageFormat::Png),
+        // _ => return Err(PortalError::Failed(format!("Failure to take screenshot"))),
+        // };
+        Ok(ScreenshotResponse::new(url))
```

Comment:
All of this seems like an implementation detail to be honest, someone implementing the portal is expected to know how to do all of this... There is no need for hand-holding here.

The goal of the demo is to be able with time to run it on the CI and test that it returns the expected responses when using ashpd's APIs or when running the ASHPD Demo locally, not as an actual impl because it depends a lot on the implementor (the compositor in for the screenshot interface)